[TensorExpr] Add another test for ExternalCalls.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52162 [TensorExpr] Add another test for ExternalCalls.**

This test demonstrates how external calls can interoperate with other
tensor computations and between themselves.

Differential Revision: [D26410813](https://our.internmc.facebook.com/intern/diff/D26410813)

Commit:
[TensorExpr] Add another test for ExternalCalls.

This test demonstrates how external calls can interoperate with other
tensor computations and between themselves.

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add another test for ExternalCalls."


This test demonstrates how external calls can interoperate with other
tensor computations and between themselves.

Differential Revision: [D26410813](https://our.internmc.facebook.com/intern/diff/D26410813)

[ghstack-poisoned]

asuhan commented on test/cpp/tensorexpr/test_external_calls.cpp:
```diff
@@ -237,5 +237,91 @@ TEST(ExternalCall, Matmul) {
   ASSERT_TRUE(at::allclose(nnc_result, ref));
 }
 
+TEST(ExternalCall, SeveralCallsUsedInEachOther) {
```
I'd like a more concise and expressive name for this test, such as `ExpressionArguments` or something similar.

ZolotukhinM replied:
How about `ComputeInterop` to underline that the test is checking how external-calls tensors interoperate with tensors built with the Compute API? I also added a comment on that.

asuhan replied:
It's ok.

Commit:
Update on "[TensorExpr] Add another test for ExternalCalls."


This test demonstrates how external calls can interoperate with other
tensor computations and between themselves.

Differential Revision: [D26410813](https://our.internmc.facebook.com/intern/diff/D26410813)

[ghstack-poisoned]
